src/parser/HLSTree.cpp

```diff
@@ -348,6 +348,7 @@ PLAYLIST::PrepareRepStatus adaptive::CHLSTree::prepareRepresentation(PLAYLIST::C
         }
 
         newSegment->range_end_ += newSegment->range_begin_;
+        newSegment->range_end_ -= 1;
```

Comment:
Why two assignemts here for the same field? Line 350 should be removed?

Reply:
No can't be removed, but could be changed to `newSegment->range_end_ += newSegment->range_begin_ - 1;`  instead of adding the new statement. I'm not a good judge of what's more correct or readable, I could go either way.

Reply:
right done

Reply:
Lol. The "-=" was not visible on my phone. It showed: newSegment->range_end_ = -1; ... Sorry for the noise.